src/compiler/commandLineParser.ts

```diff
@@ -772,6 +772,20 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
         defaultValueDescription: false,
         description: Diagnostics.Disable_emitting_comments,
     },
+    {
+        name: "noCheck",
+        type: "boolean",
+        showInSimplifiedHelpView: false,
+        category: Diagnostics.Compiler_Diagnostics,
+        description: Diagnostics.Disable_full_type_checking_only_critical_parse_and_emit_errors_will_be_reported,
+        transpileOptionValue: undefined,
+        defaultValueDescription: false,
+        affectsSemanticDiagnostics: true,
+        affectsBuildInfo: true,
+        extraValidation() {
+            return [Diagnostics.Unknown_compiler_option_0, "noCheck"];

```

Comment:
For anyone reading, this validation is what blocks this option from being passed on the CLI. This is a nicer way to do it than our usual method (just adding an entry to the `CompilerOptions` interface), since it still provides metadata for `.buildinfo` serialization and `showConfig`, plus makes it easier to test as though it were CLI-allowed, like is done in the `noCheck` unittest.